src/views/map.c

```diff
@@ -2179,6 +2180,55 @@ static gboolean _view_map_remove_pin(const dt_view_t *view, OsmGpsMapImage *pin)
   return osm_gps_map_image_remove(lib->map, pin);
 }
 
+static void track_add_point(OsmGpsMapTrack *track, OsmGpsMapPoint *point, OsmGpsMapPoint *prev_point)
+{
+  float lat, lon, prev_lat, prev_lon;
+  osm_gps_map_point_get_degrees(point, &lat, &lon);
+  osm_gps_map_point_get_degrees(prev_point, &prev_lat, &prev_lon);
+  double d, delta;
+  gboolean short_distance = TRUE;
+  if (
+    fabs(lat - prev_lat) > DT_MINIMUM_ANGULAR_DELTA_FOR_GEODESIC ||
+    fabs(lon - prev_lon) > DT_MINIMUM_ANGULAR_DELTA_FOR_GEODESIC
+  )
+  {
+    short_distance = FALSE;
+    dt_gpx_geodesic_distance(lat, lon,
+                    prev_lat, prev_lon,
+                    &d, &delta
+                  );
+  }
+
+  if ((prev_lat == 0.0 && prev_lon == 0.0) || short_distance || d < DT_MINIMUM_DISTANCE_FOR_GEODESIC)
```

Comment:
Here we have already calculated the geodesic distance.
(prev_lat == 0.0 && prev_lon == 0.0) should be evaluated and treated at the beginning of the routine.

Reply:
yes, actually I resolved the case of the first point (that was the `prev_lat == 0.0 && prev_lon == 0.0` case) out of the function